samples/bluetooth/fast_pair/locator_tag/src/dfu.c

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/settings/settings.h>
+#include <zephyr/kernel.h>
+
+#include <zephyr/bluetooth/uuid.h>
+#include <bluetooth/adv_prov.h>
+
+#include <bluetooth/services/fast_pair/fast_pair.h>
+#include <bluetooth/services/fast_pair/fmdn.h>
+
+#include <zephyr/mgmt/mcumgr/grp/os_mgmt/os_mgmt.h>
+#include <zephyr/mgmt/mcumgr/mgmt/callbacks.h>
+
+#include "app_dfu.h"
+#include "app_fp_adv.h"
+#include "app_ui.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(fp_fmdn, LOG_LEVEL_INF);
+
+/* DFU mode timeout in minutes. */
+#define DFU_MODE_TIMEOUT (1)
+
+/* UUID of the SMP service used for the DFU. */
+#define BT_UUID_SMP_SVC_VAL \
+	BT_UUID_128_ENCODE(0x8D53DC1D, 0x1DB7, 0x4CD3, 0x868B, 0x8A527460AA84)
+
+/* UUID of the SMP characteristic used for the DFU. */
+#define BT_UUID_SMP_CHAR_VAL	\
+	BT_UUID_128_ENCODE(0xda2e7828, 0xfbce, 0x4e01, 0xae9e, 0x261174997c48)
+
+#define BT_UUID_SMP_CHAR	BT_UUID_DECLARE_128(BT_UUID_SMP_CHAR_VAL)
+
+static bool fmdn_provisioned;
+static bool dfu_mode;
+
+static bool ad_enabled;
+static bool sd_enabled;
```

Comment:
We could get rid of this static variable if we follow the approach from my fp_adv.c comment